File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
    List<List<Integer>> result = new ArrayList<>(); 
    Queue<TreeNode> workList = new ArrayDeque<>();
        
    if(root == null){
        return result;
    }
    workList.add(root); 

    while(!workList.isEmpty()) { 
        List<Integer> level = new ArrayList<>(); 
        int qLength = workList.size(); 
        for(int i = 0; i < qLength; i++) {
        TreeNode currNode = workList.poll();    
        if(currNode.left != null) { workList.add(currNode.left); }
        if(currNode.right != null) { workList.add(currNode.right); } 
        level.add(currNode.val);
        }
        result.add(level); 

    }   

    return result;

        }     
    }  

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        if(not root): 
            return 0 

        count = 1 

        def dfs(curr: TreeNode, maxSF: int): 
            if(not curr):
                return

            if(curr.val > maxSF): 
                self.count += 1 
                if(curr.left): 
                    dfs(curr.left, curr.val)
                if(curr.right): 
                    dfs(curr.right, curr.val)
            else: 
                if(curr.left): 
                    dfs(curr.left, maxSF)
                if(curr.right): 
                    dfs(curr.right, maxSF)
        
        dfs(root.left, root.val)
        dfs(root.right, root.val)

        return count